cogs/cmd_interface/quiz.py

```diff
@@ -0,0 +1,170 @@
+from typing import Optional
+import typing
+
+import discord
+from discord import app_commands
+from discord.ext import commands
+
+import random
+
+keyAns = ['A. ', 'B. ', 'C. ', 'D. ', 'E.', 'F.']
+iconKey = ['🇦', '🇧', '🇨', '🇩', '🇪', '🇫']
+TOPIC_TAGS = ['Algorithms', 'Concurrency', 'Distributed Systems','Software Architecture', 'Complexity Theory']
+
+
+def createEmbed(QandA: list, choice: int = -1):
+    question = QandA[0]
+    answer = QandA[1]
+
+    correctAnswer = question.correctAnswerId - answer[0].id
+
+    embed = discord.Embed(
+        color=0xff822e,
+    )
+    embed.set_author(
+        name="Quiz: ",
+        icon_url="https://assets.leetcode.com/users/leetcode/avatar_1568224780.png"
+    )
+    embed.add_field(
+        name=question.question,
+        value="",
+        inline=False
+    )
+
+    embed.add_field(
+        name="Difficulty",
+        value=question.difficulty,
+        inline=True
+    )
+
+    embed.add_field(
+        name="Topic",
+        value=f"||{question.category}||",
+        inline=True
+    )
+
+    if choice == -1:
+        answer_view = ""
+        for i in range(len(answer)):
+            answer_view += f"```\n{keyAns[i] + answer[i].answer}\n```"
+        embed.add_field(
+            name="Answer",
+            value=answer_view,
+            inline=False
+        )
+    else:
+        answer_view = ""
+        for i in range(len(answer)):
+            if i == correctAnswer:
+                answer_view += f"```diff\n+{keyAns[i] + answer[i].answer}\n```"
+                continue
+            if i == choice:
+                answer_view += f"```diff\n-{keyAns[i] + answer[i].answer}\n```"
+                continue
+            answer_view += f"```\n{keyAns[i] + answer[i].answer}\n```"
+        embed.add_field(
+            name="Answer",
+            value=answer_view,
+            inline=False
+        )
+    embed.add_field(
+        name="Hint",
+        value="|| This is a hint ||",
+        inline=False
+    )
+    return embed
+
+
+class NavButton(discord.ui.Button['ChooseQuestion']):
```

Comment:
Nav stands for `Navigation`, but for this case, it's not navigating anything. Please change name.